File: frontend/src/types/job.ts
export interface Job {
  id: string;
  title: string;
  company: string;
  location: string;
  url: string;
  /** Direct posting URL used by the Extract JD batch. */
  job_url?: string;
  salary?: string | null;
  work_model?: string | null;
  publish_time?: string | null;
  publish_time_desc?: string | null;
  match_score?: string | null;
  description?: string | null;
  skills?: string | null;

  // -- persisted state (jobs are stored server-side now) -------------------
  source?: string;
  application_status?: string;
  /** The date the job was added, as the table shows and edits it. */
  date_added?: string;
  /** "" | "ready" | "applied" — what the Status column displays. */
  status?: string;
  /** Status cannot be chosen until a resume exists for the row. */
  hasResume?: boolean;
  /** ISO timestamp, null until the job is marked applied. */
  applied_at?: string | null;
  applied?: boolean;
  /** Nothing may act on a locked job: no extraction, no generation. */
  locked?: boolean;
  first_seen_at?: string;
  last_seen_at?: string;
}

File: backend/app/schemas/job.py
from pydantic import BaseModel


class JobListing(BaseModel):
    id: str
    title: str
    company: str
    location: str
    url: str
    salary: str | None = None
    work_model: str | None = None
    publish_time: str | None = None
    publish_time_desc: str | None = None
    match_score: str | None = None
    description: str | None = None
    skills: str | None = None


class StoredJob(BaseModel):
    """A job as held in the database, which is what the table renders."""

    id: str
    source: str = "jobright"
    title: str = ""
    company: str = ""
    location: str = ""
    url: str = ""
    description: str = ""
    salary: str = ""
    work_model: str = ""
    match_score: str = ""
    publish_time: str = ""
    publish_time_desc: str = ""
    skills: str = ""
    application_status: str = "not_applied"
    # The date the job was added, as the table shows and edits it.
    date_added: str = ""
    # "" | "ready" | "applied" -- what the Status column displays.
    status: str = ""
    # Status cannot be chosen until a resume exists for the row.
    hasResume: bool = False
    applied_at: str | None = None
    first_seen_at: str = ""
    last_seen_at: str = ""
    # Derived server-side so every caller agrees on what "applied" means.
    applied: bool = False
    # Nothing may act on a locked job: no extraction, no generation, no import
    # overwrite. Today that is exactly "applied", but the pipeline checks this
    # flag rather than the status, so a second reason to freeze a row is a
    # one-line change here.
    locked: bool = False


class ExtractSkillsRequest(BaseModel):
    description: str
    prompt: str


class ExtractSkillsResponse(BaseModel):
    skills: str
